src/server/tran_server.cpp

```diff
@@ -551,34 +551,35 @@ tran_server::connection_handler::disconnect_async (bool with_disc_msg)
 
   m_disconn_future = std::async (std::launch::async, [this, with_disc_msg]
   {
-    on_disconnecting (); // server-type specific jobs before disconnecting.
-    // m_conn is not nullptr since it's only set to nullptr here below once
-
     m_conn->stop_response_broker (); // wake up threads waiting for a response and tell them it won't be served.
```

Comment:
I think it can be removed too, because this function is also called inside the `m_conn->stop_incoming_communication_thread ()`.

Reply:
You're right. It can be much simpler. Thank you.